web/src/components/tasks/search-business-name/SearchBusinessNameForm.test.tsx

```diff
@@ -9,102 +14,126 @@ import {
 jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
 jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/data-hooks/useDocuments");
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
 jest.mock("@/lib/api-client/apiClient", () => ({
+  postBusinessFormation: jest.fn(),
+  getCompletedFiling: jest.fn(),
   searchBusinessName: jest.fn(),
 }));
+
 const mockApi = api as jest.Mocked<typeof api>;
 
-const Config = getMergedConfig();
-const onChange = jest.fn();
+describe("SearchBusinessNameForm", () => {
+  const displayContent = {
+    formationDbaContent: generateFormationDbaContent({}),
+  };
 
-describe("<SearchBusinessNameForm />", () => {
   beforeEach(() => {
     jest.resetAllMocks();
-    useMockUserData({});
-    useMockRoadmap({});
+    useSetupInitialMocks();
   });
 
-  const availableText = "AVAILABLE!";
-  const unavailableText = "UNAVAILABLE!";
-  const FakeAvailable = (): ReactElement => {
-    return <div>{availableText}</div>;
-  };
-  const FakeUnavailable = (): ReactElement => {
-    return <div>{unavailableText}</div>;
+  const getPageHelper = async (initialProfileData?: Partial<ProfileData>): Promise<FormationPageHelpers> => {
+    const profileData = generateFormationProfileData({
+      ...initialProfileData,
+      legalStructureId: "limited-liability-company",
+      businessPersona: "FOREIGN",
+    });
+    const formationData = {
+      formationFormData: generateFormationFormData(
+        {},
+        {
+          legalStructureId: castPublicFilingLegalTypeToFormationType(
+            profileData.legalStructureId as PublicFilingLegalType,
+            profileData.businessPersona
+          ),
+        }
+      ),
+      formationResponse: undefined,
+      getFilingResponse: undefined,
+      completedFilingPayment: false,
+      businessNameAvailability: undefined,
+      dbaBusinessNameAvailability: undefined,
+      lastVisitedPageIndex: 0,
+    };
+    const page = preparePage(generateUserData({ profileData, formationData }), displayContent);
+    return page;
   };
 
-  const renderForm = (): void => {
-    render(
-      <SearchBusinessNameForm
-        available={FakeAvailable}
-        unavailable={FakeUnavailable}
-        onChange={onChange}
-        config={{
-          searchButtonText: Config.searchBusinessNameTask.searchButtonText,
-          searchButtonTestId: "search-availability",
-        }}
-      />
-    );
-  };
+  it("displays modal when legal structure Edit button clicked", async () => {
+    const page = await getPageHelper();
+    await page.searchBusinessName({ status: "AVAILABLE" });
+
+    expect(true).toBe(true);
+  });
 
-  it("pre-fills the text field with the business name entered in onboarding", () => {
-    useMockProfileData({ businessName: "Best Pizza" });
-    renderForm();
+  it("pre-fills the text field with the business name entered in onboarding", async () => {
```

Comment:
maybe "business name in profile" might be more descriptive now that onboarding doesnt ask for Business Name anymore

Reply:
updated